entsoe-util/src/main/java/com/powsybl/entsoe/util/MergedXnodeImpl.java

```diff
@@ -39,17 +51,30 @@ public MergedXnodeImpl(Line line, float rdp, float xdp, double xnodeP1, double x
         super(line);
         this.rdp = checkDividerPosition(rdp);
         this.xdp = checkDividerPosition(xdp);
+        this.line1Name = Objects.requireNonNull(line1Name);
         this.xnodeP1 = checkPowerFlow(xnodeP1);
         this.xnodeQ1 = checkPowerFlow(xnodeQ1);
+        this.line2Name = Objects.requireNonNull(line2Name);
         this.xnodeP2 = checkPowerFlow(xnodeP2);
         this.xnodeQ2 = checkPowerFlow(xnodeQ2);
-        this.line1Name = Objects.requireNonNull(line1Name);
-        this.line2Name = Objects.requireNonNull(line2Name);
         this.code = Objects.requireNonNull(code);
     }
 
-    private static float checkDividerPosition(float dp) {
-        if (dp < 0f || dp > 1f || Double.isNaN(dp)) {
+    public MergedXnodeImpl(Line line, float rdp, float xdp,
+                           String line1Name, boolean line1Fictitious, double xnodeP1, double xnodeQ1, float b1dp, float g1dp,
+                           String line2Name, boolean line2Fictitious, double xnodeP2, double xnodeQ2, float b2dp, float g2dp,
+                           String code) {
+        this(line, rdp, xdp, xnodeP1, xnodeQ1, xnodeP2, xnodeQ2, line1Name, line2Name, code);
+        this.line1Fictitious = line1Fictitious;
+        this.b1dp = checkDividerPosition(b1dp);
+        this.g1dp = checkDividerPosition(g1dp);
+        this.line2Fictitious = line2Fictitious;
+        this.b2dp = checkDividerPosition(b2dp);
+        this.g2dp = checkDividerPosition(g2dp);
+    }
+
+    private float checkDividerPosition(float dp) {
```

Comment:
For me, the correct implementation should be:
```java
if (Double.isNaN(dp)) {
       return 0.5f;
}
if (dp < 0f || dp > 1f) {
       throw new IllegalArgumentException("Invalid divider postion: " + dp);
}
return dp;
```

Reply:
Done